contracts/src/BeefyClient.sol

```diff
@@ -390,49 +411,27 @@ contract BeefyClient {
         }
     }
 
-    /**
-     * @dev Calculate minimum number of required signatures for the current validator set.
-     *
-     * This function approximates f(x) defined below for x in [1, 21_846):
-     *
-     *  x <= 10: f(x) = x * (2/3)
-     *  x  > 10: f(x) = max(10, ceil(log2(3 * x))
-     *
-     * Research by W3F suggests that `ceil(log2(3 * x))` is a minimum number of signatures required to make an
-     * attack unfeasible. We put a further minimum bound of 10 on this value for extra security.
-     *
-     * If the session has less than 10 active validators it's definitely some sort of local testnet
-     * and we use different logic (minimum 2/3 + 1 validators must sign).
-     *
-     * One assumption is that Polkadot/Kusama will never have more than roughly 20,000 active validators in a session.
-     * As of writing this comment, Polkadot has 300 validators and Kusama has around 1000 validators,
-     * so we are well within those limits.
-     *
-     * In any case, an order of magnitude increase in validator set sizes will likely require a re-architecture
-     * of Polkadot that would make this contract obsolete well before the assumption becomes a problem.
-     *
-     * Constants generated with the help of scripts/minsigs.py
-     */
-    function minimumSignatureThreshold(uint256 validatorSetLen) internal pure returns (uint256) {
-        if (validatorSetLen <= 10) {
+    function minSignatures(uint256 validatorSetLen, uint256 signatureUseCount) internal pure returns (uint256) {
+        if (validatorSetLen <= BASE_SIGNATURE_COUNT) {
             return validatorSetLen - (validatorSetLen - 1) / 3;
-        } else if (validatorSetLen < 342) {
-            return 10;
-        } else if (validatorSetLen < 683) {
-            return 11;
-        } else if (validatorSetLen < 1366) {
-            return 12;
-        } else if (validatorSetLen < 2731) {
-            return 13;
-        } else if (validatorSetLen < 5462) {
-            return 14;
-        } else if (validatorSetLen < 10923) {
-            return 15;
-        } else if (validatorSetLen < 21846) {
-            return 16;
-        } else {
-            return 17;
         }
+
+        if (signatureUseCount == 0) {
+            return BASE_SIGNATURE_COUNT;
+        }
+
+        // log2 rounded down is the index of the highest bit
+        uint256 log2SignatureUse = Bits.highestBitSet(signatureUseCount);
+
+        // We need to get the value of log2 rounded up. To do this we can compare if signatureUseCount is
+        // greater than the signatureUseCount is larger than its most significant bit. If it is larger then
```

Comment:
This comment looks like it needs to be reformatted. for example, there is an unnecessary newline after `then`.

Reply:
Resolved in https://github.com/Snowfork/snowbridge/pull/956/commits/dc5f1d2d416e80572aedaa9d054ee9ba52feb123